kms-message/src/kms_response_parser.c

```diff
@@ -348,7 +364,16 @@ kms_response_parser_get_response (kms_response_parser_t *parser)
 int
 kms_response_parser_status (kms_response_parser_t *parser)
 {
-   if (!parser || !(parser->response)) {
+   if (!parser) {
+      return 0;
+   }
+   
+   if (parser->kmip) {
+      KMS_ERROR (parser, "kms_response_parser_status not applicable to KMIP");
```

Comment:
Confirming: if `kms_response_parser_status()` is not applicable to KMIP, is the only means of querying error status via `kms_response_parser_error()`? Can a user distinguish a KMS response parser error from a KMS KMIP response parser error (if they need to)?

Reply:
> is the only means of querying error status via kms_response_parser_error()

Yes, `kms_response_parser_error` returns an error during parsing the response.

`kms_response_parser_status` returns the HTTP status code. I think `kms_response_parser_status` would be better as `kms_http_response_get_status` on the `kms_response_t`. I added a comment to kms_response_parser.h to clarify.

> Can a user distinguish a KMS response parser error from a KMS KMIP response parser error (if they need to)?

Yes, `kms_kmip_response_parser_feed` and `kms_kmip_response_parser_get_response` include `KMIP` as part of the error message.